batch/batch/server/server.py

```diff
@@ -556,7 +562,7 @@ def to_dict(self):
             'job_id': self.job_id,
             'state': self._state
         }
-        if self._state == 'Complete':
+        if self.is_complete():
```

Comment:
Should this ever be false? In this function it doesn't seem like there is any incomplete state: you mark and exit_code by line 510.

Reply:
Yes. This is returning the JSON object to the client and can be called at any time when the user calls status or wait.

Reply:
Ah right, so while on line 540 we have `await self.set_state(new_state)`, but the state could have changed in between that call and line 565?

I could see that, although I wonder if in that case we don't want to force the state to be complete. I may not be thinking about this correctly.

Reply:
Line 565 is a different function than mark_complete where the state is being updated. I think it is definitely possible to return from `to_dict` with something out of date. However, it is not possible for a single request to have an inconsistent state because each request creates a new job object. Not sure if this makes sense.